apps/frontend/src/libs/components/icon/icon.tsx

```diff
@@ -0,0 +1,27 @@
+/// <reference types="vite-plugin-svgr/client" />
+
+import { type FC } from "react";
+
+import Eye from "~/assets/images/icons/eye.svg?react";
+import StrikedEye from "~/assets/images/icons/striked-eye.svg?react";
+
+type IconName = "eye" | "strikedEye";
+
+const iconNameToSvg: Record<IconName, FC<React.SVGProps<SVGSVGElement>>> = {
+	eye: Eye,
```

Comment:
I think `IconName` and `iconNameToSvg` should be moved from this file. That's because, they will be changed in the future (cause we will need to add new icons), and if we keep them here, there will be a possibility to break Icon component. So without them Icon component will be fully (at least for now) done and there will be no problem to safely and easily add new icons.

Reply:
Where do you think is the best place to store them?

Reply:
For type `IconName`, the best I think would be:
`apps/frontend/src/libs/components/icon/types/types.ts` and `apps/frontend/src/libs/components/icon/types/icon-name.type.ts`
And for `iconNameToSvg` I'm not sure. But maybe good place can be:
`apps/frontend/src/libs/components/icon/helpers/helpers.ts` and `apps/frontend/src/libs/components/icon/helpers/icon-name-to-svg.helper.ts`

Reply:
Oh, about `iconNameToSvg`. I mean that you should move whole object and not introduce that identity function ;)
So the `icon-name-to-svg.helper.ts` should looks like
```ts
/// <reference types="vite-plugin-svgr/client" />

import { type FC } from "react";

import Eye from "~/assets/images/icons/eye.svg?react";
import StrikedEye from "~/assets/images/icons/striked-eye.svg?react";

import { type IconName } from "../types/types.js";

const iconNameToSvg: Record<IconName, FC<React.SVGProps<SVGSVGElement>>> = {
	eye: Eye,
	strikedEye: StrikedEye,
};

export { iconNameToSvg };
```
and then you can import and use this map (i.e. `iconNameToSvg`) inside `icon.tsx` file like:

```ts
import { iconNameToSvg } from "./helpers/helpers.js";
import { type IconName } from "./types/types.js";

type Properties = {
  className?: string;
  name: IconName;
  onClick?: () => void;
};

const Icon = ({ className, name, onClick }: Properties): JSX.Element => {
  const SvgComponent = iconNameToSvg[name];
  
  return <SvgComponent className={className} onClick={onClick} />
};

export { Icon };
```

Reply:
Also there is no need for `Icon` type, so in `icon.type.ts` file. To keep project consistent, you better keep here `type Properties` (as you can see in my code snippet), and also take a look at component declaration. It must looks like: `const Icon = ({ className, name, onClick }: Properties): JSX.Element => {`.